pkg/stub/config.go

```diff
@@ -67,94 +67,93 @@ func (h *Handler) SpecValidation(cfg *v1.Config) error {
-			if beforeError && !afterError {
-				clearImageImportErrors = true
-			}
+	streamsThatWereSkipped := map[string]bool{}
+	unskippedStreams := map[string]bool{}
+	// build the list of skipped imagestreams; assumes buildSkipFilters called beforehand
+	for _, stream := range cfg.Status.SkippedImagestreams {
+		streamsThatWereSkipped[stream] = true
+		// if something that was skipped (in Status list) is no longer skipped, build list/capture that fact
+		// this should capture reductions in the skipped list, as well as changing entries, but the number of skips
+		// staying the same
+		if _, ok := h.skippedImagestreams[stream]; !ok {
+			unskippedStreams[stream] = true
+			configChangeAtAll = true // set to true in case sizes are equal, but contents have changed
+			configChangeRequireUpsert = true
+			streamChange = true
 		}
 	}
-	if changeInContent {
-		return changeInContent, changeInContent, clearImageImportErrors
+
+	if streamChange {
+		logrus.Printf("SkippedImagestreams changed from %#v to %#v", cfg.Status.SkippedImagestreams, cfg.Spec.SkippedImagestreams)
 	}
 
-	if len(cfg.Spec.SkippedTemplates) != len(cfg.Status.SkippedTemplates) {
-		logrus.Printf("SkippedTemplates changed from %#v to %#v", cfg.Status.SkippedTemplates, cfg.Spec.SkippedTemplates)
-		if len(cfg.Spec.SkippedTemplates) < len(cfg.Status.SkippedTemplates) {
-			return true, true, false
-		}
-		for _, tpl := range cfg.Status.SkippedTemplates {
-			// even if the skipped list has been increased, if a tpl we were skipping
-			// has been removed, we need to upsert; assumes buildSkipFilters called beforehand
-			if _, ok := h.skippedTemplates[tpl]; !ok {
-				return true, true, false
-			}
+	// see if need to update the cfg from the main loop to clear out any prior image import
+	// errors for skipped streams
+	for _, stream := range cfg.Spec.SkippedImagestreams {
+		importErrors := cfg.Condition(v1.ImportImageErrorsExist)
+		beforeError := cfg.NameInReason(importErrors.Reason, stream)
+		importErrors = h.clearStreamFromImportError(stream, cfg.Condition(v1.ImportImageErrorsExist), cfg)
+		afterError := cfg.NameInReason(importErrors.Reason, stream)
+		if beforeError && !afterError {
+			clearImageImportErrors = true
+			// we do not break here cause we want to clear out all possible streams
 		}
-		return true, false, false
 	}
 
-	for i, skip := range cfg.Spec.SkippedTemplates {
-		if skip != cfg.Status.SkippedTemplates[i] {
-			logrus.Printf("SkippedTemplates changed in content from %s to %s", cfg.Status.SkippedTemplates[i], skip)
-			return true, true, false
-		}
+	if cfg.Spec.SamplesRegistry != cfg.Status.SamplesRegistry {
+		logrus.Printf("SamplesRegistry changed from %s to %s", cfg.Status.SamplesRegistry, cfg.Spec.SamplesRegistry)
+		configChangeAtAll = true
+		configChangeRequireUpsert = true
+		// skip/unskip imagestream does not matter, reset that list
+		registryChange = true
+		unskippedStreams = map[string]bool{}
```

Comment:
could do this up front and immediately return, to avoid all the skipped/unskipped processing, right?

Reply:
(presumably we also need to reset any import errors when that happens)

Reply:
no I changed this and moved it from up top because I realized that the registry change does not affect templates

that said, I could do this after the template skip processing and before the imagestream skip processing, 
but then I liked the idea of
- logging the skip changes for dbg
- having only one return statement in this function

But I don't feel super strongly on those 2 points.  Would you like me to move this between the template skips and imagestream skips ?

Reply:
I went ahead and made this change while making the one above